apps_product_creator/models/product_product.py

```diff
@@ -0,0 +1,214 @@
+    )
+    image_small = fields.Binary(
+        "Small-sized image",
+        compute='_compute_images',
+        inverse='_set_image_small',
+        help="Image of the product variant (Small-sized image of "
+             "product template if false).",
+    )
+    image_medium = fields.Binary(
+        "Medium-sized image",
+        compute='_compute_images',
+        inverse='_set_image_medium',
+        help="Image of the product variant (Medium-sized image of "
+             "product template if false).",
+    )
+
+    @api.model
+    def create(self, values):
+        """
+        Apply the behaviour to product used to save modules.
+        :param values: dict
+        :return: self recordset
+        """
+        self._manage_product_module(values)
+        return super(ProductProduct, self).create(values)
+
+    @api.model
+    def _manage_product_module(self, values):
+        """
+        Manage product variants related to modules.
+        During the creation of a new product.product (who is a variant), we
+        check if the product is used to save a module (by the odoo_module_id
+        field).
+        Then check into given variant attributes if we find the variant used
+        to save the module version (with XML ID:
+        apps_product_creator.attribute_odoo_version).
+        If it's the case, we check the version (based on the attribute value)
+        and just link the product.product to this version
+        :param values: dict
+        :return: bool
+        """
+        product_tmpl_key = "product_tmpl_id"
+        attr_key = "attribute_value_ids"
+        check_keys = [
+            product_tmpl_key,
+            attr_key,
+        ]
+        values_keys = values.keys()
+        # Check if mandatory keys (for product module) are given
+        if all([k in values_keys for k in check_keys]):
+            product_obj = self.env['product.template']
+            product = product_obj.browse(values.get(product_tmpl_key, []))
+            # If the product template is used to save an Odoo module
+            if self._check_related_to_module(product):
+                attribute = self._get_version_attribute(values)
+                version = self._get_version_with_attribute(
+                    product.odoo_module_id.module_version_ids, attribute)
+                # If we don't have a result, the ID will be False
+                values.update({
+                    'odoo_module_version_id': version.id,
+                })
+        return True
+
+    @api.model
+    def _get_version_with_attribute(self, versions, attribute):
+        """
+        Get the version recordset (odoo.module.version) into the given version
+        (versions) based on the attribute name.
+        :param versions: odoo.module.version recordset
+        :param attribute: product.attribute.value recordset
+        :return: odoo.module.version recordset
+        """
+        # Attribute value (so the name field) must be the milestone name
```

Comment:
I am not comfortable using milestones. Our repos and versions are based on branches, not milestones (barely used)

Reply:
In this module we use the milestone related to a branch, simply as a shortcut to get the version of that branch.
This choice is also due to the technical constraint imposed by the module github_connector_odoo which uses the model github.organization.milestone for keeping track of the versions available for any module.

I'm not perfectly happy with this choice neither but at the moment I don't see a better solution. You have suggestions on how to get the branch version on a branch, based on data provided by the github-connector modules?

Reply:
Then we need to bring the topic to @legalsylvain before it is too late.
Basically we should be able to create versions of the module based on the branch it is located.

Reply:
I don't understand exactly the problem. Is it just a name problem ? or it is a feature problem ? 
milestone are just some branches. for exemple (l10n_brazil branch ) : 
![image](https://user-images.githubusercontent.com/3407482/31510625-547a0d74-af85-11e7-82c5-59e91c3296d8.png)

6.0 / 6.1 / 7.0 / 8.0 / 9.0 / 10.0 / 11.0 are branches and milestones (I mean, a stable and usable revision)
gh-pages is a branch, but not a milestone.

In a project with many repositories, like OCA, milestone are defined by the organization. (7, 8, 9...) for that reason, i added the model github.organization.milestone. 

regards.

Reply:
Hi all, 

Please do not rely on the milestone to determine the odoo series for which an addon applies. Use the branch prefix (the part before the first dash), or the version in the manifest (the first two digits, reliable from version 8 onwards).

We must leave the liberty to maintainers to use milestones for what they are made for, ie organizing and planning future releases.

Reply:
Again, I'm not using github milestone. (no synchronisation). I'm using "some" branches. That is the objective of github.organization.milestone. (to define which branches are relevant). 

the previous name of this model was github.organization.serie and community asked to change the name of the model and milestone was OK. 

https://github.com/OCA/interface-github/pull/2/commits/a96d476e1b6d38f59aee9909ec29ed3bd4b50d79#diff-ae93b445bcca140d9d99caa33763c37f

Reply:
Thanks for the clarification @legalsylvain, I was confused by the use of the milestone name. Series was much more clear to me, sorry I missed that discussion.

Reply:
> Series was much more clear to me,

Me too ! ;-)

The discussion is here.
https://github.com/OCA/interface-github/pull/1#discussion_r93159698

@lasley, @elicoidal  : could you give your feedback, and choose a name that is OK for all ? 

regards.

Reply:
@legalsylvain @sbidoul 
Originally, series is a notion which does not exist in github so I suggested to stick to github naming.
Milestones are currently not useful in our organization because they are not consistently used.
We need to stick to the branch name or as suggested by @sbidoul the module manifest.

I would suggest interface-github module to highlight the branch the module is in (whatever the milestone) and apps-store module to use the assumption that one branch is one version (with a crappy text compare to the internal version model to rule out WIP branches).
I am open to improve this: suggestions welcome!

Reply:
@elicoidal *organization level* milestone is not a github concept either AFAICT. We are trying to model an odoo specific concept here.

Reply:
@sbidoul then maybe I am confused with the naming here. I thought about Milestones as defined here after:
![screenshot from 2017-10-13 11-59-29](https://user-images.githubusercontent.com/7600613/31541241-4d047992-b00e-11e7-886f-b8dc006e7e4d.png)

Reply:
@elicoidal these are project level milestones and not synchronized by github_connector. See also https://github.com/OCA/interface-github/pull/1#discussion_r144515098.

Reply:
@sbidoul OK: I have never seen other milestones in the interface for the organization.
As said: any name is good to me as long it is consistent.
As for using them for the module version, we would agree that this is probably not the best option

Reply:
please note: https://github.com/OCA/interface-github/pull/2#issuecomment-336509906